buildSrc/src/main/groovy/org/gradle/testing/performance/generator/tasks/AbstractProjectGeneratorTask.groovy

```diff
@@ -30,27 +36,62 @@ import org.gradle.testing.performance.generator.TestProject
  * Original tangled mess of a project generator.
  */
 abstract class AbstractProjectGeneratorTask extends ProjectGeneratorTask {
+
+    @Input
     int sourceFiles = 1
+    @Input
     int projectCount = 1
+    @Input
+    @Optional
     Integer testSourceFiles
+    @Input
     int linesOfCodePerSourceFile = 5
+    @Input
     int filesPerPackage = 100
+    @Input
+    int numberOfExternalDependencies = 0
+
+    @Nested
+    final DependencyGraph dependencyGraph = new DependencyGraph()
+
+    @Input
     List<String> additionalProjectFiles = []
-    final List<TestProject> projects = []
+
+    @Internal
+    String buildSrcTemplate
+    @Internal
     List<String> rootProjectTemplates = ['root-project']
+    @Internal
     List<String> subProjectTemplates = ['project-with-source']
```

Comment:
How are these inputs tracked?

Reply:
The actual content of used templates is tracked here: https://github.com/gradle/gradle/pull/3606/files/ad2f09a2182ec43f13767d8c2bc69cfa30d59b40#diff-3df88e2eceb4da23ddeed01dbb250481R84

Reply:
Roger. You can add this info as the `value()` of the `@Internal` annotation.